forwards_paper.tex

```diff
@@ -611,18 +629,327 @@ \subsection*{Tree sequence simplification}
+Since every chromosome inherits material from both parents,
+without simplification this would produce tables of
+$2NT$ nodes and
+$4NT$ edges.
+Suppose that $T$ is large enough that all samples coalesce within the simulation with high probability,
+($T \sim 20N$, say).
+After simplification, we are left with the tree sequence describing the history
+of only the current generation of $2N$ individuals.
+This tree sequence has $4N-2$ edges to describe the leftmost tree;
+and each time the marginal tree changes along the sequence,
+four edges end and four new edges begin 
+(except for changes affecting the root, which require fewer; see \citet{kelleher2016efficient}).
+Coalescent theory tells us that
+the expected total length of the edges of a marginal tree is approximately $4N\log(2N)$,
+which is also equal to the mean number of ancestral recombination events that occur on a branch of the marginal tree,
+since we have taken one crossover per individual per generation.
+Not all such recombinations actually change the tree topology,
+so four times this gives us an upper bound on the expected number of additional edges.
+Similarly, not every new edge derives from a never-before-seen node,
+but the number of nodes is at most equal to the number of edges plus the sample size.
+With $T=20N$, this reduces the initial storage of $120 N^2$ items to 
+% 4N - 2 + 2N + 4N log(2N) * 4 * 2 = 6N + 32 N log(2N)
+$6N + 32 N \log(2N)$;
+at $N=10^4$, a factor of 3,700.
+
+%% Progressive memory reduction as a function of T
+To get an idea of how required space depends on the length of time between simplification steps,
+suppose instead that we have run a simulation of $2N$ individuals for $T$ generations,
+begun with no prior history.
+How many of the resulting $4NT$ edges are required after simplification?
+As above, the expected number of edges is bounded by $2N-2$ plus four times the mean marginal tree length.
+Roughly, the length of time for which a marginal tree has $k$ tips is
+$2N/k(k-1) = 2N(1/(k-1) - 1/k)$,
+and so the time to go from $N$ to $n$ tips is $2N(1/(n-1) - 1/(N-1))$.
+This implies that after running for time $T$ we expect to
+have around $n(T)$ roots, where $n(T) \approx 1 + 2N/(T+2)$.
+The total tree length over this time is
+$2 N \sum_{k=n(T)+1}^{N-1} 1/k$, which
+% is approximately
+% \begin{align*}
+%     2 N \log\left( \frac{N}{1 + 2 N/(T+2)} \right) .
+% \end{align*}
+leads to an upper bound on the number of edges of
+\begin{align}
+    \label{eqn:edge_bound}
+    2 N \left( 1 + 4 \log\left( \frac{NT}{T + 2 N} \right)\right) .
+\end{align}
+This holds up quite well in practice,
+as shown in Figure~\ref{fig:simplify_complexity}.
+% by comparison to
+% the number of edges actually required for 50 independent Wright--Fisher simulations,
+% as a function of time (obtained from Algorithm~\algref{W} with $s=1$)
+% is shown in Figure~\ref{fig:simplify_complexity}, compared to this prediction.
+
+\begin{figure}
```

Comment:
Here I've just added two other subfigures.  As placeholders, obviously, to see if this is how/where we want to display that information.  What do you think?

Reply:
Good idea I think. Maybe we should just keep it to two figures though, and get rid of (C)? This keeps the context to forwards time simulations. It might be confusing to suddenly switch in a figure based on a coalescent sim. Plus it'll look better with two probably.

Reply:
Hm, but we do refer to (C) in the text.  And besides, we could motivate it by saying "suppose we do a large fwds simulation, then simplified it down to everyone (and so get sthg equiv to a coalescent sim), how fast to subset out a collection of samples?"

Reply:
OK, not let's keep it.

Reply:
Hah; the email I got says

> OK, not let's keep it.

I'm glad the comment online makes more sense. =)

Reply:
Yeah, hooray for editable comments!